File: 2008/1/A.py
#!/usr/bin/env python3

import sys

def dotp(xs, ys):
    return sum([x * y for (x,y) in zip(xs,ys)])

def solve(xs, ys):
    return dotp(sorted(xs), sorted(ys, reverse=True))

def main():
    ifp = sys.stdin
    ofp = sys.stdout

    T = int(ifp.readline())
    for i in range(1, T+1):
        n  = ifp.readline()
        xs = map(int, ifp.readline().split())
        ys = map(int, ifp.readline().split())
        sol = solve(xs, ys)
        print("Case #{}: {}".format(i, sol), file=ofp)
        ofp.flush()
        
if __name__ == "__main__":
    main()

File: 2008/1/A.cc

#include <iostream>
using namespace std;

int64_t solve( int64_t *x,  int64_t *y, int n) {
  sort(x, x+n);
  sort(y, y+n, greater<int>());

  int64_t dotp = 0;
  for(int i = 0; i < n; i++) {
    dotp += x[i] * y[i];
  }
    
  return dotp;
}



int main(int argc, char** argv) {

  int T;
  cin >> T;

  int64_t x[800];
  int64_t y[800];

  for(int t = 1; t <= T; t++) {
    int n;
    cin >> n;

    for(int i = 0; i<n; i++) {
      cin >> x[i];
    }
    for(int i = 0; i<n; i++) {
      cin >> y[i];
    }

    int64_t sol = solve(x, y, n);
    cout << "Case #" << t << ": " << sol << endl;
  }
}
